qutebrowser/completion/models/sortfilter.py

```diff
@@ -63,23 +61,18 @@ def set_pattern(self, val):
 
         Invalidates the filter and re-sorts the model.
 
-        If the current completion model overrides sort(), it is used.
-        If not, the default implementation in QCompletionFilterModel is called.
-
         Args:
             val: The value to set.
         """
         with debug.log_time(log.completion, 'Setting filter pattern'):
+            # empty value clears cache (not necessary for correctness, but
+            # helps with keeping memory requirements relatively low)
+            if not val:
+                self.srcmodel.filtered_out_cache = {}
             self.pattern = val
-            val = re.escape(val)
-            val = val.replace(r'\ ', r'.*')
-            self.pattern_re = re.compile(val, re.IGNORECASE)
             self.invalidateFilter()
             sortcol = 0
-            try:
-                self.srcmodel.sort(sortcol)
-            except NotImplementedError:
-                self.sort(sortcol)
+            self.sort(sortcol)
             self.invalidate()
```

Comment:
I'm not using `completion -> shrink` myself, but at quick glance repeating https://github.com/The-Compiler/qutebrowser/commit/b5781f0ed3ad2fdd517afa37f7845919aa319c99 on top of current master doesn't seem to break anything. The completion widget shrinks and I haven't noticed any weird behaviour.